master/buildbot/process/workerforbuilder.py

```diff
@@ -221,7 +222,8 @@ def attached(self, worker, commands):
         return super().attached(worker, commands)
 
     def substantiate(self, build):
-        return self.worker.substantiate(self, build)
+        if self.worker:
+            return self.worker.substantiate(self, build)
```

Comment:
I think we should keep appropriate return when the condition is not matched, probably `defer.succeed(None)`

Reply:
Good catch.
Used `defer.succeed(False)` since `worker.substantiate` returns a `Deferred[bool]`.